docs/architecture/blazor-for-web-forms-developers/pages-routing-layouts.md

```diff
@@ -0,0 +1,259 @@
+---
+title: Pages, routing, and layouts
+description: How to create pages in Blazor, work with client-side routing, and manage page layouts.
+author: danroth27
+ms.author: daroth
+ms.date: 09/19/2019
+---
+
+# Pages, routing, and layouts
+
+ASP.NET Web Forms apps are made up of pages defined in *.aspx* files. The address of each page is based on its file path in the project. When a browser makes a request to the page, the contents of the page are dynamically rendered on the server based on the page's HTML markup and server controls.
+
+In Blazor, each page in the app is a component, typically defined in a *.razor* file, with one or more specified routes. Routing mostly happens client-side without involving a specific server request. The browser first makes a request to the root address of the app. A root `Router` component in the Blazor app then handles intercepting navigation requests and them to the correct component.
+
+Blazor also supports deep linking, where the browser makes a request to a specific route other than the root of the app. Requests for deep links sent to the server are routed to the Blazor app, which then routes the request client-side to the correct component.
+
+A simple page in ASP.NET Web Forms might look like this:
+
+*Name.aspx*
+
+```html
+<%@ Page Title="Name" Language="C#" MasterPageFile="~/Site.Master" AutoEventWireup="true" CodeBehind="Name.aspx.cs" Inherits="WebApplication1.Name" %>
+
+<asp:Content ID="BodyContent" ContentPlaceHolderID="MainContent" runat="server">
+    <div>
+        What is your name?<br />
+        <asp:TextBox ID="TextBox1" runat="server"></asp:TextBox>
+        <asp:Button ID="Button1" runat="server" Text="Submit" OnClick="Button1_Click" />
+    </div>
+    <div>
+        <asp:Literal ID="Literal1" runat="server" />
+    </div>
+</asp:Content>
+```
+
+*Name.aspx.cs*
+```csharp
+public partial class Name : System.Web.UI.Page
+{
+    protected void Button1_Click1(object sender, EventArgs e)
+    {
+        Literal1.Text = "Hello " + TextBox1.Text;
+    }
+}
+```
+
+The equivalent page in a Blazor app would look like this:
+
+*Name.razor*
+
+```razor
+@page "/Name"
+@layout MainLayout
+
+<div>
+    What is your name?<br />
+    <input @bind="text" />
+    <button @onclick="OnClick">Submit</button>
+</div>
+<div>
+    if (name != null)
+    {
+        Hello @name
+    }
+</div>
+
+@code {
+    string text;
+    string name;
+
+    void OnClick() {
+        name = text;
```

Comment:
Don't you need the `NotifyStateChanged` or whatever that method is called?

Reply:
Not from an event handler. We assume the state has probably changed after a UI event occurs.